### PR TITLE
Enable remix singleFetch

### PR DESCRIPTION
**Pull request by aaschlote**
### Changes

Enable [singleFetch](https://remix.run/docs/en/main/guides/single-fetch) due remix json has been deprecated in the last version. Right now we have more than 50 maintenance issues in Sonar related it.

image

### CODE REVIEW

File: app/routes/action.set-analytics.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const action = async ({ request }: ActionFunctionArgs) => {
   const headers = await consentCookieFromRequest({ request });
 
   if (clientJavaScriptAvailable) {
-    return json({ success: true }, { headers });
+    return data({ success: true }, { headers: headers });
```
[the remark this answers is not in the document]
Comment (aaschlote) [reply app/routes/action.set-analytics.tsx:14]: we also can do:

``new Response(null, { headers: headers }}``

The content of the response body is not important

**Review by Spencer6497 — APPROVED** [review]
thanks for getting this working!

File: app/routes/action.set-analytics.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chohner) [reply app/routes/action.set-analytics.tsx:14]: the `headers: headers`  change shouldn't be necessary, right?

File: vite.config.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { defineConfig } from "vite";
 import tsconfigPaths from "vite-tsconfig-paths";
 import { envOnlyMacros } from "vite-env-only";
 
-installGlobals();
+installGlobals({ nativeFetch: true });
```
Comment (chohner) [inline vite.config.ts:8]: I thought we can completely remove the `installGlobals`, since we are using nodes native fetch? not 100% sure thought..

**Review by chohner — APPROVED** [review]
lgtm, but could you change the PR title to "enable singleFetch"?

would be nice if the change was so easy after all :)

File: app/routes/action.set-analytics.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaschlote) [reply app/routes/action.set-analytics.tsx:14]: yeap, it's not needed, removing it!

File: vite.config.ts (hunk 2 above)
Comment (aaschlote) [reply vite.config.ts]: it seems we don't need it all. Do you know why it was added it? 

Btw, I removed it. Thanks for the hint!

**Pull request merged**